From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 27 Aug 2018 00:29:35 +0100
Subject: string: Avoid ABI change in 4.9.114
Forwarded: not-needed

Commit 5c067898febc "string: drop __must_check from strscpy() ..." is
considered by genksyms to be an ABI change, but it's not.  Hide the
change from genksyms.

---
--- a/include/linux/string.h
+++ b/include/linux/string.h
@@ -27,7 +27,11 @@ extern char * strncpy(char *,const char
 size_t strlcpy(char *, const char *, size_t);
 #endif
 #ifndef __HAVE_ARCH_STRSCPY
+#ifndef __GENKSYMS__
 ssize_t strscpy(char *, const char *, size_t);
+#else
+ssize_t __must_check strscpy(char *, const char *, size_t);
+#endif
 #endif
 #ifndef __HAVE_ARCH_STRCAT
 extern char * strcat(char *, const char *);
